### PR TITLE
[MCC-176237] Code refactoring and version up to 1.4.2

### PR DESCRIPTION
**Pull request by junshao**
- Refactored ComponentInfoHandler and added unit tests for it.
- Included the fix reported by @echen-mdsol in https://github.com/junshao/Medidata.Cloud.Thermometer.RaveCommon/pull/24. Thanks @echen-mdsol 
- Updated Medidata.Cloud.Thermometer to 1.6.0
- Version up to 1.4.2.

@echen-mdsol Please review and merge if OK. Thanks.

### CODE REVIEW

File: Mct.RaveCommon/ComponentInfoHandler.cs [hunk 1]
```diff
@@ -13,50 +14,54 @@ namespace Medidata.Cloud.Thermometer.RaveCommon
...
+        [ExcludeFromCodeCoverage]
+        internal virtual Assembly GetComponentAssembly()
         {
-            var assemblies = AppDomain.CurrentDomain.GetAssemblies();
-
-            var assembly = assemblies.FirstOrDefault(a => RaveComponentNames.Contains(a.GetName().Name));
-
+            var assmeblies = AppDomain.CurrentDomain.GetAssemblies();
+            var assembly = assmeblies.FirstOrDefault(a => RaveComponentNames.Contains(a.GetName().Name));
             return assembly;
```
Comment (art2003) [inline Mct.RaveCommon/ComponentInfoHandler.cs]: Do we need to check for CompareOptions.IgnoreCase

File: Mct.RaveCommon/ComponentInfoHandler.cs [hunk 2]
```diff
@@ -13,50 +14,54 @@ namespace Medidata.Cloud.Thermometer.RaveCommon
...
-            };
-        }
-
-        protected internal virtual IEnumerable<string> RaveComponentNames { get; private set; }
+            "Medidata.Core.Service",
+            "Medidata.Rave.Integration.Service",
+            "Medidata.RaveWebServices.Web",
+            "MedidataRAVE",
+            "RaveCrystalViewer"
+        };
```
Comment (art2003) [inline Mct.RaveCommon/ComponentInfoHandler.cs]: Maybe we can create Enum

Comment (junshao) [inline Mct.RaveCommon/ComponentInfoHandler.cs]: They are strings, cannot be Enum, as cannot use "." as Enum name nor string as Enum value.

File: Mct.RaveCommon/ComponentInfoHandler.cs (hunk 1 above)
Comment (junshao) [inline Mct.RaveCommon/ComponentInfoHandler.cs]: They are assembly names, which are so always case sensitive.

**Comment by art2003** [comment]
Ok, thanks @junshao. Merging

**Pull request merged**